service/src/test/java/uk/gov/hmcts/reform/fpl/service/cmo/UploadCMOServiceTest.java

```diff
@@ -0,0 +1,427 @@
+
+        DynamicList expected = dynamicList(hearings.get(0).getId(), hearings.get(1).getId(), hearings.get(2).getId());
+
+        assertThat(dynamicList).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldBuildADynamicListWithASelectedValue() {
+        List<Element<HearingBooking>> hearings = hearings();
+
+        DynamicList dynamicList = service.buildDynamicList(hearings, hearings.get(0).getId());
+
+        DynamicList expected = dynamicList(hearings.get(0).getId(),
+            hearings.get(1).getId(),
+            hearings.get(2).getId(),
+            true);
+
+        assertThat(dynamicList).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldReturnCorrectPageDataWhenThereAreMultipleHearings() {
+        List<Element<HearingBooking>> hearings = hearings();
+
+        Map<String, Object> initialPageData = service.getInitialPageData(hearings, List.of());
+
+        Map<String, Object> expected = Map.of(
+            "pastHearingList", dynamicList(hearings.get(0).getId(), hearings.get(1).getId(), hearings.get(2).getId()),
+            "numHearings", "MULTI"
+        );
+
+        assertThat(initialPageData).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldReturnCorrectPageDataWhenThereAreMultipleHearingsWithSomeAlreadyAssigned() {
+        List<Element<HearingBooking>> hearings = new ArrayList<>(hearings());
+
+        Element<CaseManagementOrder> cmo = element(CaseManagementOrder.builder().build());
+        Element<HearingBooking> hearing = element(
+            hearing(CASE_MANAGEMENT, LocalDateTime.of(2020, 1, 15, 0, 0), cmo.getId())
+        );
+
+        hearings.add(hearing);
+
+        Map<String, Object> initialPageData = service.getInitialPageData(hearings, List.of(cmo));
+
+        Map<String, Object> expected = Map.of(
+            "pastHearingList", dynamicList(hearings.get(0).getId(), hearings.get(1).getId(), hearings.get(2).getId()),
+            "numHearings", "MULTI",
+            "multiHearingsWithCMOs", "Case management hearing, 15 January 2020",
+            "showHearingsMultiTextArea", "YES"
+        );
+
+        assertThat(initialPageData).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldReturnCorrectPageDataWhenThereIsOneRemainingHearing() {
+        List<Element<HearingBooking>> hearings = List.of(element(
+            hearing(CASE_MANAGEMENT, LocalDateTime.of(2020, 2, 1, 0, 0))
+        ));
+
+        Map<String, Object> initialPageData = service.getInitialPageData(hearings, List.of());
+
+        Map<String, Object> expected = Map.of(
+            "numHearings", "SINGLE",
+            "cmoHearingInfo", "Send agreed CMO for Case management hearing, 1 February 2020."
+                + "\nThis must have been discussed by all parties at the hearing.",
+            "cmoJudgeInfo", "His Honour Judge Dredd"
+        );
+
+        assertThat(initialPageData).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldReturnCorrectPageDataWhenThereIsOneRemainingHearingWithSomeAlreadyAssigned() {
```

Comment:
Assigned seems a bit off for me, am I missing something? It's late so I might be!

Reply:
nah, names are hard and they probably could be clearer

Reply:
assigned in this case meant had a CMO Id attached I believe. But as you can see I am not sure what it meant a couple of days after and I wrote the test so it probably needs a rename